pkg/cli/mirror/operator.go

```diff
@@ -472,16 +563,40 @@ func (o *OperatorOptions) plan(ctx context.Context, dc *declcfg.DeclarativeConfi
 			return nil, err
 		}
 	} else {
+		// ctlgDir is the result of converting targetCtlg.Ref to <repoPath> example: foo/bar/baz/image/sha256:XXXX
 		ctlgDir, err := operator.GenerateCatalogDir(targetCtlg.Ref)
 		if err != nil {
 			return nil, err
 		}
+		// layoutDir looks like <some path>/src/catalogs/<repoPath>/layout
+		// this will be the destination of the copy action that follows
 		layoutDir := filepath.Join(o.Dir, config.SourceDir, config.CatalogsDir, ctlgDir, config.LayoutsDir)
 		if err := os.MkdirAll(layoutDir, os.ModePerm); err != nil {
 			return nil, fmt.Errorf("error catalog layout dir: %v", err)
 		}
-		if err := copy.Copy(v1alpha2.TrimProtocol(ctlgRef.OCIFBCPath), layoutDir); err != nil {
-			return nil, fmt.Errorf("error copying oci fbc catalog to layout directory: %v", err)
+		// obtain the source directory for the OCI content
+		ociSourcePath := v1alpha2.TrimProtocol(ctlgRef.OCIFBCPath)
+
+		// Now copy the individual components of the source OCI source to its layout dir destination.
+		// This is done to ensure that files/folders that are not part of the OCI layout specification
+		// are not copied.
+
+		// copyOCILayoutFileOrFolder will copy a file or folder that belongs to a oci layout
+		copyOCILayoutFileOrFolder := func(sourcePath, destinationPath, fileOrDir string) error {
```

Comment:
One of the goals of future refactoring is to reduce the quantity of anonymous functions since they make the code less readable and it is difficult to follow when debug is required. 

Is it possible to define this func outside of plan and call it as a defined func?

Reply:
Moved to standalone function

Reply:
Thanks @jchunkins